File: AdmitadSDKDemo/ObjCDemoApp/RegistrationVC.m
//
//  RegistrationVC.m
//  fff
//
//  Created by Dmitry Cherednikov on 08/11/2017.
//  Copyright © 2017 Tachos. All rights reserved.
//

#import "RegistrationVC.h"
#import "AppDelegate.h"
#import "UserInfo.h"

@import AdmitadSDK;

@interface RegistrationVC () {

    IBOutlet __weak UILabel *userInfoLabel;
    IBOutlet __weak UIBarButtonItem *goToCartBarButtonItem;
    IBOutlet __weak UIButton *registerButton;

    AdmitadTracker *admitadTracker;
}

@end

@implementation RegistrationVC

- (void)viewDidLoad {

    [super viewDidLoad];

    admitadTracker = [AdmitadTracker sharedInstance];

    self.title = @"Registration";
    userInfoLabel.text = [NSString stringWithFormat:@"Register user %@\n with id: %@", [UserInfo userName], [UserInfo userId]];

    goToCartBarButtonItem.enabled = NO;
}

- (IBAction)registerButtonPressed {

    [admitadTracker trackRegisterEventWithUserId:[UserInfo userId] completion:^(AdmitadError *error) {

        AppDelegate *appDelegate = (AppDelegate *)[UIApplication sharedApplication].delegate;

        if (!error) {

            goToCartBarButtonItem.enabled = YES;
            registerButton.enabled = NO;
            [appDelegate showAlertWithMessage: @"Register event is successfully tracked"];
        }
        else {

            [appDelegate showAlertWithMessage: @"Error tracking Register event"];
        }
    }];
}

@end

File: AdmitadSDKDemo/SwiftDemoApp/RegistrationVC.swift
//
//  RegistrationVC.swift
//  DeeplinkSDKTest
//
//  Created by Dmitry Cherednikov on 24/10/2017.
//  Copyright © 2017 tachos. All rights reserved.
//

import Foundation
import UIKit
import AdmitadSDK

class RegistrationVC: UIViewController {
    @IBOutlet weak var userInfoLabel: UILabel!
    @IBOutlet weak var goToCartBarButtonItem: UIBarButtonItem!
    @IBOutlet weak var registerButton: UIButton!

    private var admitadTracker = AdmitadTracker.sharedInstance
}

// MARK: - UIViewController lifecycle
extension RegistrationVC {
    override func viewDidLoad() {
        super.viewDidLoad()
        title = "Registration"
        userInfoLabel.text = "Register user \(UserInfo.name)\n with id: \(UserInfo.id)"
        goToCartBarButtonItem.isEnabled = false
    }
}

// MARK: - actions
extension RegistrationVC {
    @IBAction func registerButtonPressed() {
        admitadTracker.trackRegisterEvent(userId: UserInfo.id) { [unowned self] (error) in
            if error != nil {
                return
            }
            self.goToCartBarButtonItem.isEnabled = true
            self.registerButton.isEnabled = false
        }
    }
}
